lib/cryptoexchange/exchanges/gate/services/market.rb

```diff
@@ -0,0 +1,50 @@
+module Cryptoexchange::Exchanges
+  module Gate
+    module Services
+      class Market < Cryptoexchange::Services::Market
+        class << self
+          def supports_individual_ticker_query?
+            false
+          end
+        end
+
+        def fetch
+          output = super ticker_url
+          adapt_all output
+        end
+
+        def ticker_url
+          "#{Cryptoexchange::Exchanges::Gate::Market::API_URL}/tickers"
+        end
+
+        def adapt_all(output)
+          output.map do |market_pair_key, ticker|
+            base, target = market_pair_key.split('_')
```

Comment:
From http://data.gate.io/api2/1/tickers
API returns market_pair_key of `btc_usdt`
I believe we want it to be `target, base = market_pair_key.split('_')`
base is USDT, target is BTC

Reply:
`target, base` correction